src/Groups/matrices/MatGrp.jl

```diff
@@ -547,6 +547,53 @@ function order(::Type{T}, G::MatrixGroup) where T <: IntegerUnion
    return T(res)::T
 end
 
+"""
+    map_entries(f, G::MatrixGroup)
+
+Return the matrix group obtained by applying `f` element-wise to
+each generator of `G`.
+
+`f` can be a ring or a field, a suitable map, or a Julia function.
+
+# Examples
+```jldoctest
+julia> mat = matrix(ZZ, 2, 2, [1, 1, 0, 1]);
+
+julia> G = matrix_group(mat);
+
+julia> G2 = map_entries(x -> -x, G)
+Matrix group of degree 2
+  over integer ring
+
+julia> is_finite(G2)
+false
+
+julia> order(map_entries(GF(3), G))
+3
+```
+"""
+function map_entries(R::Ring, G::MatrixGroup)
```

Comment:
There is also `change_base_ring` for this case, which does the same. Maybe one should call the other?

Reply:
I think one could just have one method of the form
```
map_entries(f, G::MatrixGroup)
```
without specifying the type of the first argument. This is how all other `map_*` methods usually are implemented.

Reply:
@joschmitt `change_base_ring` seems to only cover the case where the first argument is a ring, right? Of course we can make sure one is implemented in terms of the other. The current method is
```julia
function change_base_ring(R::Ring, G::MatrixGroup)
  g = dense_matrix_type(R)[]
  for h in gens(G)
    push!(g, map_entries(R, h.elm))
  end
  return matrix_group(g)
end
```
which does not deal with empty generator lists. We could just change it to
```julia
change_base_ring(R::Ring, G::MatrixGroup) = map_entries(R, G)
```

@thofma those other `map_entries` methods don't need to deal with an empty list of matrices, though... But yeah one could merge the methods though one could consider the two simple methods as "optimizations" of the generic `Function` one? Is there any harm in having these additional methods?

Reply:
No, you are right. Maybe my suggestion is just to remove the restriction on the first argument and shuffle the docstring, so that it is not at the "wrong" method (at the moment the signature of the docstring and the function are not the same).

Reply:
> We could just change it to
> 
> ```julia
> change_base_ring(R::Ring, G::MatrixGroup) = map_entries(R, G)
> ```

Yes, exactly.